forms/FormField.php

```diff
@@ -20,13 +20,15 @@
  * For example, data might be saved to the filesystem instead of the data record, or saved to a
  * component of the data record instead of the data record itself.
  *
+ * A form field can be represented as structured data through {@link FormSchema},
+ * including both structure (name, id, attributes, etc.) and state (field value).
+ * Can be used by for JSON data which is consumed by a front-end application.
+ *
  * @package forms
  * @subpackage core
  */
 class FormField extends RequestHandler {
 
-	use SilverStripe\Forms\Schema\FormFieldSchemaTrait;
```

Comment:
OK, so this comment is about this entire commit but I have to say that this is quite surprising. I don't see the fact that the trait is used in 1 place as a reason not to break the code into smaller pieces.

Perhaps the trait should mark `abstract function getSchemaDataType()`, and the implementation of only that method is pulled into `FormField`?